OneCommunityGlobal/HighestGoodNetworkApp
# Olga tool details page

# Description

Components for Individual tool display based on the below Figma design:

![Screen Shot 2024-01-04 at 12 00 43 PM](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/107726913/44bde7f5-3c0e-46c9-bdf0-ad8afa4cd4b8)

## Related PRS (if any):
This frontend PR is related to the [#684](https://github.com/OneCommunityGlobal/HGNRest/pull/684) backend PR.


## Main changes explained:

-created components: ToolModal.jsx, ToolDetailPage.jsx
-added ToolDetailPage.css for styling
-added tool actions, reducer and route


## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. check into the backend branch from a related backend PR and start the backend
4. go to http://localhost:3000/bmdashboard/tools/657b3a082d8cc6bd67704ea2 to view a tool with 'Purchase' status
5. go to http://localhost:3000/bmdashboard/tools/657b3a082d8cc6bd67704ea3 to view a tool with 'Rental' status
6. Check that the links are working. Link to Buy/Rent takes you to a new page (HomeDepot website at the moment). Description link opens a modal with tool description.
7. Check that if the tools Ownership is Purchase, Rental Duration line is not displayed.
8. Check the page in different screensizes.


## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/107726913/fc60a335-0d13-40c8-abf8-b2b46d3e1277
